chore: update for mrk support in keystore (#267)
awslabs/aws-encryption-sdk-specification: framework/branch-key-store.md

A valid [AWS KMS Key ARN](./aws-kms/aws-kms-key-arn.md#a-valid-aws-kms-arn)
that wraps and unwraps keys stored in Amazon DynamoDB.

The KMS Configuration MUST distinguish between strict ARN compatibility and MRK ARN compatibility.

> Both compatibility modes are allowed with both mrk ARN's and non-mrk ARNs.

Reviewer: ```suggestion
Both compatibility modes are allowed with both mrk ARN's and single region ARNs.
```

> #### AWS Key Compatibility

Reviewer: ```suggestion
#### AWS Key ARN Compatibility
```

> For two ARNs to be compatible :

Reviewer 2: ```suggestion
For two ARNs to be compatible:
```

> If the [AWS KMS Configuration](#aws-kms-configuration) designates strict ARN compatibility,

Reviewer: ```suggestion
If the [AWS KMS Configuration](#aws-kms-configuration) designates single region ARN compatibility,
```

The call to AWS KMS GenerateDataKeyWithoutPlaintext MUST use the configured AWS KMS client to make the call.
The operation MUST call AWS KMS GenerateDataKeyWithoutPlaintext with a request constructed as follows:

> - `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore

Reviewer 2: very minor nit:

```suggestion
- `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```

The call to AWS KMS GenerateDataKeyWithoutPlaintext MUST use the configured AWS KMS client to make the call.
The operation MUST call AWS KMS GenerateDataKeyWithoutPlaintext with a request constructed as follows:

> - `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore

Reviewer 2: ```suggestion
- `KeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```

with a request constructed as follows:

- `SourceEncryptionContext` MUST be the [DECRYPT_ONLY encryption context for branch keys](#decrypt_only-encryption-context).

> - `SourceKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore

Reviewer 2: ```suggestion
- `SourceKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```

- `CiphertextBlob` MUST be the wrapped DECRYPT_ONLY Branch Key.
- ReEncrypt `GrantTokens` MUST be this keystore's [grant tokens](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#grant_token).

> - `DestinationKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore

Reviewer 2: ```suggestion
- `DestinationKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured `AWS KMS Key ARN` in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
```
